numpyro/distributions/constraints.py

```diff
@@ -96,6 +98,14 @@ def feasible_like(self, prototype):
         return jax.numpy.zeros_like(prototype)
 
 
+class _Circular(Constraint):
+    def __call__(self, x):
+        return (x >= -math.pi) & (x < math.pi)
+
+    def feasible_like(self, prototype):
+        return jax.numpy.broadcast_to(math.pi, jax.numpy.shape(prototype))
```

Comment:
maybe use 0.0 instead, since `math.pi` doesn't strictly satisfy the constraint, and `-math.pi` is at a boundary

Reply:
That makes sense!